src/components/Spawn/SpawnEffect.component.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC, useEffect, useMemo, useRef } from 'react';
+import { AnimationMixer, Group, LoopRepeat } from 'three';
+import { useFrame, useGraph } from '@react-three/fiber';
+import { triggerCallback, useGltfLoader } from '../../services';
+import { SpawnState } from '../Avatar/Avatar.component';
+
+interface SpawnEffectProps {
+  onSpawnEffectFinish: () => void;
+  onMountEffect: SpawnState['onMountEffect'];
+}
+export const SpawnEffect: FC<SpawnEffectProps> = ({ onMountEffect, onSpawnEffectFinish }) => {
+  const ref = useRef<Group>(null);
+
+  const [effectRunning, setEffectRunning] = React.useState(true);
+
+  const { scene: mountEffectScene } = useGltfLoader(onMountEffect?.src || '');
```

Comment:
We should avoid possibility that input is invalid and the `Loader` is triggered which will cause an error.

Reply:
Yes, checked in the parent and here we modify the type to be always an string (and validated in previous step Spawn component)